### PR TITLE
TELCODOCS-1013 - workload partitioning for SNO

### PR DESCRIPTION
**Pull request by aireilly**
https://issues.redhat.com/browse/TELCODOCS-1013

Merge to main, CP to 4.11+

Preview: 
* [Moved existing WP content under the reference SNO section](https://53376--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.html#ztp-workload-partitioning-sno_sno-configure-for-vdu)
* [Updated WP CR reference section](https://53376--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-reference-cluster-configuration-for-vdu.html#ztp-sno-du-enabling-workload-partitioning_sno-configure-for-vdu)

### CODE REVIEW

File: snippets/ztp-performance-profile.adoc [hunk 1]
```diff
@@ -30,6 +30,11 @@ spec:
 <1> Ensure that the value for `name` matches that specified in the `spec.profile.data` field of `TunedPerformancePatch.yaml` and the `status.configuration.source.name` field of `validatorCRs/informDuValidator.yaml`.
 <2> Configures UEFI secure boot for the cluster host.
 <3> Set the isolated CPUs. Ensure all of the Hyper-Threading pairs match.
++
+[NOTE]
+====
+The reserved and isolated CPU pools must not overlap and together must span all available cores.
```
Comment (pixelsoccupied) [inline snippets/ztp-performance-profile.adoc:36]: Can we add extra bit of wording to be extra careful as leaving out any core causes an undefined behaviour? And add the "Important" tag like here https://docs.openshift.com/container-platform/4.10/scalability_and_performance/cnf-performance-addon-operator-for-low-latency-nodes.html#cnf-cpu-infra-container_cnf-master ?

File: snippets/ztp-example-siteconfig.adoc [hunk 2]
```diff
@@ -43,7 +43,7 @@ spec:
         bootMode: "UEFI" <10>
         rootDeviceHints:
           wwn: "0x11111000000asd123"
-        cpuset: "0-3"  <11>
+        cpuset: "0-1,52-53"  <11>
```
Comment (pixelsoccupied) [inline snippets/ztp-example-siteconfig.adoc:46]: To active workload partitioning  for SNO the must much `cpuset` of PerformanceProfile with PGT and `reserved` of crio-o with SiteConfig (and make sure that the CPU cores selected are not arbitrary e.g selected NUMA cores in this case for max performance)

 Is there something like mentioned is this PR?

Comment (aireilly) [reply snippets/ztp-example-siteconfig.adoc:46]: We mention this explicitly here: https://github.com/openshift/openshift-docs/pull/53376/commits/5334027d7016582c7c054c3ff31369e48d316827#diff-86d1f3820339a335de3b823ffe6eac402324cc3541cebe1069ecd2dc7075f695R11

Updated slightly. PTAL

File: snippets/ztp-performance-profile.adoc (hunk 1 above)
Comment (aireilly) [reply snippets/ztp-performance-profile.adoc]: updated

File: modules/ztp-enabling-workload-partitioning-sno.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+====
+For maximum performance, ensure that the `reserved` and `isolated` CPU sets do not share CPU cores across NUMA zones.
+====
+
+* The workload partitioning `MachineConfig` CR pins the {product-title} infrastructure pods to a defined `cpuset` configuration.
+* The `PerformanceProfile` CR pins the systemd services to the reserved CPUs.
+
+[IMPORTANT]
+====
+The value for the `reserved` field specified in the `PerformanceProfile` custom resource (CR) must match the `cpuset` field in the workload partitioning `MachineConfig` CR.
```
Comment (adellape) [inline modules/ztp-enabling-workload-partitioning-sno.adoc:24]: Looks like the first spelling-out of "custom resource (CR)" for the module is down here now. Maybe swap it with the first "CR" from earlier ("`SiteConfig` CR")?

**Comment by aireilly** [comment]
@adellape nuts! Sorry! LMK if you want to undo the merge/CPs

**Pull request merged**